metriken-exposition/src/snapshot.rs

```diff
@@ -0,0 +1,53 @@
+use crate::*;
+
+/// Contains a snapshot of metric readings.
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Snapshot {
```

Comment:
Given that you're exposing every single field on this struct via a method what about just making the fields public and marking this struct as `#[non_exhaustive]`?

Modifying the types of any of the fields here is likely to result in a major version bump anyway.

Reply:
👍 seems okay for now. I'm not sure we want to allow mutable access into the internals. Exposing the fields via methods gives us better control. But until it's a problem, marking the fields pub.